tests/backend/test_bosonic_backend.py

```diff
@@ -0,0 +1,560 @@
+        prog_0H = sf.Program(1)
+        with prog_0H.context as q0:
+            sf.ops.GKP(epsilon=eps) | q0[0]
+            sf.ops.Rgate(np.pi/2) | q0[0]
+              
+        backend_0H = bosonic.BosonicBackend()
+        backend_0H.run_prog(prog_0H)
+        state_0H = backend_0H.state()
+        wigner_0H = state_0H.wigner(0,x,p)
+        
+        # Prepare GKP +
+        prog_plus = sf.Program(1)
+        with prog_plus.context as qp:
+            sf.ops.GKP(state=[np.pi/2,0],epsilon=eps) | qp[0]
+              
+        backend_plus = bosonic.BosonicBackend()
+        backend_plus.run_prog(prog_plus)
+        state_plus = backend_plus.state()
+        wigner_plus = state_plus.wigner(0,x,p)
+        
+        assert np.allclose(wigner_0H,wigner_plus)
+    
+    @pytest.mark.parametrize("eps", EPS_VALS)
+    def test_gkp_state_parity(self,eps):
+        r"""Checks that GKP states yield the right parity."""    
+        prog = sf.Program(1)
+        with prog.context as q:
+            sf.ops.GKP(epsilon=eps) | q[0]
+              
+        backend = bosonic.BosonicBackend()
+        backend.run_prog(prog)
+        state = backend.state()
+        assert np.allclose(state.parity_expectation([0]),1,atol=r_fock)
+    
+    def test_gkp_complex(self):
+        r"""Checks that trying to call a complex representation
+        raises an error.
+        
+        This test can be updated once the complex representation is
+        implemented."""   
+        with pytest.raises(NotImplementedError):
+            prog = sf.Program(1)
+            with prog.context as q:
+                sf.ops.GKP(desc="complex") | q[0]
+                  
+            backend = bosonic.BosonicBackend()
+            backend.run_prog(prog)
+
+class TestBosonicUserSpecifiedState():
+    """Checks the Bosonic preparation method."""
+    
+    @pytest.mark.parametrize("alpha", ALPHA_VALS)
+    def test_complex_weight(self, alpha):
+        r"""Checks that Bosonic creates a state with user-specifed weights, means
+        and covariances."""
+        dummy_backend = bosonic.BosonicBackend()
+        dummy_backend.begin_circuit(1)
+        # Get weights, means and covs for a cat state
+        weights, means, covs = dummy_backend.prepare_cat(alpha,0,0,'complex',0)
+        
+        # Initiate the state, but use Bosonic method
+        backend = bosonic.BosonicBackend()
+        prog = sf.Program(1)
+        with prog.context as q1:
+            sf.ops.Bosonic(weights,means,covs) | q1[0]
+        backend.run_prog(prog)
+        state = backend.state()
+        
+        #Prepare using the Catstate method
+        backend2 = bosonic.BosonicBackend()
+        prog2 = sf.Program(1)
+        with prog2.context as q2:
+            sf.ops.Catstate(alpha) | q2[0]
+        backend2.run_prog(prog2)
+        state2 = backend2.state()
+        
+        #Check the two states are equal
+        assert state.__eq__(state2)
+        
+class TestBosonicPrograms():
```

Comment:
No need for parentheses if there is no inheritance.
```suggestion
class TestBosonicPrograms:
```